modules/siddhi-extensions/output-mappers/wso2event-output-mapper/src/main/java/org/wso2/siddhi/extension/output/mapper/wso2event/WSO2EventOutputMapper.java

```diff
@@ -13,64 +16,62 @@
 @Extension(
         name = "wso2event",
         namespace = "outputmapper",
-        description = ""
+        description = "Event to WSO2Event output mapper."
 )
 public class WSO2EventOutputMapper extends OutputMapper {
     private StreamDefinition streamDefinition;
-    Map<String, String> options;
+    private OptionHolder optionHolder;
     private static final String PROPERTY_META_PREFIX = "meta_";
     private static final String PROPERTY_CORRELATION_PREFIX = "correlation_";
     private static final String STREAM_ID = "streamID";
-    private boolean isDynamicStreamID;
 
     /**
      * Initialize the mapper and the mapping configurations.
      *
      * @param streamDefinition       The stream definition
-     * @param options                Additional mapping options
-     * @param unmappedDynamicOptions Unmapped dynamic options
+     * @param optionHolder           Option holder containing static and dynamic options
+     * @param payloadTemplateBuilder Unmapped payload for reference
      */
     @Override
-    public void init(StreamDefinition streamDefinition, Map<String, String> options, Map<String, String> unmappedDynamicOptions) {
+    public void init(StreamDefinition streamDefinition, OptionHolder optionHolder, TemplateBuilder payloadTemplateBuilder) {
         this.streamDefinition = streamDefinition;
-        this.options = options;
+        this.optionHolder = optionHolder;
+    }
 
-        //streamID is expected to receive as an option/dynamic option
-        List<String> availableOptions = new ArrayList<>();
-        availableOptions.addAll(options.keySet());
-        availableOptions.addAll(unmappedDynamicOptions.keySet());
-        //validate for mandatory field streamID
-        if (availableOptions.contains(STREAM_ID)) {
-            isDynamicStreamID = unmappedDynamicOptions.containsKey(STREAM_ID);
-        } else {
-            throw new ExecutionPlanCreationException(String.format("{{%s}} configuration " +
-                    "could not be found in provided configs.", STREAM_ID));
+    /**
+     * Map and publish the given {@link Event} array
+     *
+     * @param events                  Event object array
+     * @param outputTransportCallback output transport callback
+     * @param optionHolder            option holder containing static and dynamic options
+     * @param payloadTemplateBuilder  Unmapped payload for reference
+     */
+    @Override
+    public void mapAndSend(Event[] events, OutputTransportCallback outputTransportCallback,
+                           OptionHolder optionHolder, TemplateBuilder payloadTemplateBuilder)
+            throws ConnectionUnavailableException {
+        //TODO add support to publish multiple events
+        for (Event event : events) {
```

Comment:
Each loop iteration may throw ConnectionUnavailableException checked Exception therefore, I think it's better to keep the for-loop as it is. WDYT?